### PR TITLE
Generify commit method of WritableDatabase instances and add moveToPartition method. 

### PR DESCRIPTION
**Pull request by dickensc**
This is a part of the changes made in the larger onlinePSL project.

Generify commit method of WritableDatabase instances to support committing ObservedAtoms. This is used to support partial grounding methods.

Add moveToPartition method which will move a predicate from an old partition to a specified partition. This is also used to support partial grounding methods.

### CODE REVIEW

File: psl-core/src/main/java/org/linqs/psl/database/rdbms/RDBMSDatabase.java [hunk 1]
```diff
@@ -18,22 +18,18 @@
...
 import org.linqs.psl.database.Database;
 import org.linqs.psl.database.DatabaseQuery;
 import org.linqs.psl.database.Partition;
 import org.linqs.psl.database.ResultList;
 import org.linqs.psl.database.QueryResultIterable;
-import org.linqs.psl.database.atom.AtomCache;
 import org.linqs.psl.model.atom.GroundAtom;
-import org.linqs.psl.model.atom.ObservedAtom;
 import org.linqs.psl.model.atom.QueryAtom;
 import org.linqs.psl.model.atom.RandomVariableAtom;
```
Comment (eriq-augustine) [inline psl-core/src/main/java/org/linqs/psl/database/rdbms/RDBMSDatabase.java:28]: Can this import also be dropped?

**Review by eriq-augustine — CHANGES_REQUESTED** [review]
Just one question.

File: psl-core/src/main/java/org/linqs/psl/database/rdbms/RDBMSDatabase.java (hunk 1 above)
Comment (dickensc) [reply psl-core/src/main/java/org/linqs/psl/database/rdbms/RDBMSDatabase.java:28]: Yes this is dropped in online PSL branch as well, if that is what you are asking?

**Comment by eriq-augustine** [comment]
lgtm!

**Pull request merged**